jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -167,7 +167,8 @@ public class HttpParser
             for (HttpHeader h : HttpHeader.values())
             {
                 HttpField httpField = new HttpField(h, UNMATCHED_VALUE);
-                map.put(httpField.toString(), httpField);
+                map.put(h + ": ", httpField);
+                map.put(h + ":", httpField);
```

Comment:
Do we really want to cache the no-space-after-column case? 
It's bad formatting by the client, so it deserves less efficiency?

Reply:
I added it in because I initially write a test that was just looking up "Content-Type:" (no space) and was surprised by the miss.     But I will remove for now to just fix the issue and we can think about other optimizations separately.